posts/viewsfolder/feed_views.py

```diff
@@ -34,16 +35,42 @@ def get_feed(self):
         return allPosts
 
     def get(self, request):
+        # The Idea
+        # although not bothering with defining source and origin idk
+        # (1) get our local posts: source == us, origin == us
+        # (2) for each server s we communicated with:
+        #           get all public posts they have (including servers they communicate with)
+        #           for each post p they give us:
+        #               p.source <- s
+        #               p.origin == p.origin (ie DO NOT CHANGE IT)
+        # (3) serve back all the above posts to the original requester
         user = request.user
-        posts = self.get_feed()
+
+        # serve_images = preferences.SitePreferences.serve_others_images
+        # if serve_images:
+        #     local_posts = Post.objects.filter(visibility='PUBLIC').order_by("-published")
+        # else:
+        #     local_posts = Post.objects.filter(visibility='PUBLIC').exclude(
+        #         contentType__in=['img/png;base64', 'image/jpeg;base64'])
+
+        local_posts = Post.objects.filter(visibility='PUBLIC').order_by("-published").exclude(
+            unlisted=True
+        )
+
+        local_posts_list = list(local_posts)
+        # NOTE: not necessarily false but assuming false for now
+        # TODO: do I need to filter out image posts based on site preferences for mr worldwide too? because rn its not
+        worldwide_posts_list = mr_worldwide(user, False)
+        posts = local_posts_list + worldwide_posts_list
+
         serializer = PostSerializer(posts, many=True)
         contentTypes = []
         for post in posts:
             if post.contentType == "text/markdown":
                 contentTypes.append(commonmark.commonmark(post.content))
             else:
                 contentTypes.append("<p>" + escape(post.content) + "</p>")
-        return render(request, 'post/public-make-post.html',
```

Comment:
idk what this file was

Reply:
Probably some weird merge conflict thing